In [1]:
import numpy as np
from mostlyai import MostlyAI

In [2]:
# initialize client
mostly = MostlyAI(api_key='mostly-adcd77a4ca7523519dc19b4ec9c9a7f6e0b2e37e16599b4703b03944dc824ed2', base_url='https://app.mostly.ai')

In [3]:
# fetch configuration via API
g = mostly.generators.get('1d9f85fd-277c-40d8-985f-02d3f39c948b')
config = g.config()
config

{'name': 'SK Patient Data',
 'description': None,
 'tables': [{'name': 'SK Patient Data',
   'sourceConnectorId': '2f880c16-adf9-4b89-85a6-29bb9316c741',
   'location': 'mostly-production-mostly-s3-bucket-mostly-app/connectors/2f880c16-adf9-4b89-85a6-29bb9316c741/SK Patient Data/cleaned_sk_data.csv',
   'data': None,
   'modelConfiguration': {'maxSampleSize': 6712,
    'batchSize': None,
    'modelSize': 'M',
    'maxTrainingTime': 10,
    'maxEpochs': 100,
    'maxSequenceWindow': 100,
    'enableFlexibleGeneration': False,
    'valueProtection': False,
    'rareCategoryReplacementMethod': 'CONSTANT',
    'compute': 'd5f093a6-208f-4b79-92c6-6d50d02d8c8c'},
   'textModelConfiguration': None,
   'primaryKey': None,
   'foreignKeys': None,
   'columns': [{'name': 'latitude', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'longitude', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'sex', 'modelEncodingType': 'NUMERIC_DISCRETE'},
    {'name': 'age', 'modelEncodingType': 'NUMERI

In [4]:
# probe the generator for some samples
mostly.probe('1d9f85fd-277c-40d8-985f-02d3f39c948b', size=10)

latitude  longitude  sex  age  state
0  0.642826   0.314941    1    5      0
1  0.956715  -0.670758    0    6      0
2  0.949465  -0.786510    1    2      0
3  0.665888  -0.517780    0    5      0
4  0.820518  -0.828993    0    2      0
5  0.004652  -1.198811    0    6      0
6  0.886566  -0.785633    0    2      0
7 -1.592439   1.154750    0    4      0
8  0.809267   1.610390    1    4      0
9  0.054896  -0.590495    0    4      0

We need 20 synthetic data sets. Original data set size was 6712, so we need $6712 \times 20 = 134240$ samples.

In [5]:
# use generator to create a synthetic dataset
sd = mostly.generate('1d9f85fd-277c-40d8-985f-02d3f39c948b', size=134_240)
sd.data()

Created synthetic dataset ]8;id=877074;https://app.mostly.ai/d/synthetic-datasets/8ce55f6a-68d2-45dd-8487-27a7854940c6\8ce55f6a-68d2-45dd-8487-27a7854940c6]8;;\ with generator ]8;id=950177;https://app.mostly.ai/d/generators/1d9f85fd-277c-40d8-985f-02d3f39c948b\1d9f85fd-277c-40d8-985f-02d3f39c948b]8;;\

Started synthetic dataset generation

Output()

🎉 Your synthetic dataset is ready! Use it to consume the generated data. Share it so others can do the same.

latitude  longitude  sex  age  state
0     0.764661  -1.239943    1    3      0
1     0.814318  -0.628751    1    4      0
2    -1.324501   0.963262    1    2      0
3     0.828459  -0.691455    0    2      0
4      -0.7394   0.825086    1    2      0
...        ...        ...  ...  ...    ...
8385  0.929708  -0.799815    0    6      0
8386 -0.129979    0.92899    0    1      0
8387 -1.710391   1.598611    0    2      0
8388  0.064244  -0.581221    0    4      0
8389  0.874804   -0.86633    0    2      0

[33560 rows x 5 columns]

Split the generated synthetic data into 20 data sets.

In [6]:
split_sd = np.array_split(sd.data(), 20)

In [7]:
for i, sX in enumerate(split_sd):
    sX.to_csv("../Data/SK/Synthetic Datasets/mostly_" + str(i) + ".csv", index=False)